extensions/amp-story-education/0.1/test/test-amp-story-education.js

```diff
@@ -174,4 +182,69 @@ describes.realWin('amp-story-education', {amp: true}, env => {
       expect(storyEducation.containerEl_).to.have.attribute('hidden');
     });
   });
+
+  describe('amp-story-education viewer messaging', () => {
+    it('should not send viewer message if not visible', async () => {
+      const sendStub = env.sandbox.stub(viewer, 'sendMessageAwaitResponse');
+      // Viewer is not visible.
+      env.sandbox
+        .stub(storyEducation.getAmpDoc(), 'whenFirstVisible')
+        .rejects();
+
+      storyEducation.buildCallback();
+      await Promise.resolve();
+
+      expect(sendStub).to.not.have.been.called;
+    });
+
+    it('should send canShowScreens for navigation on build', async () => {
+      const sendStub = env.sandbox.stub(viewer, 'sendMessageAwaitResponse');
+      env.sandbox
+        .stub(storyEducation.getAmpDoc(), 'whenFirstVisible')
+        .resolves();
+
+      storyEducation.buildCallback();
+      await Promise.resolve(); // Microtask tick.
+
+      expect(sendStub).to.have.been.calledOnceWith('canShowScreens', {
+        screens: [{screen: 'on'}],
+      });
+    });
+
+    it('should show navigation screen on viewer response', async () => {
+      env.sandbox.stub(viewer, 'sendMessageAwaitResponse').resolves({
+        screens: [{screen: 'on', show: true}],
+      });
+      env.sandbox
+        .stub(storyEducation.getAmpDoc(), 'whenFirstVisible')
+        .resolves();
+
+      storyEducation.buildCallback();
+      await Promise.resolve(); // whenFirstVisible icrotask tick.
+      await Promise.resolve(); // sendMessageAwaitResponse microtask tick.
+
+      const navigationTapEl = storyEducation.containerEl_.querySelector(
+        '[step="tap"]'
+      );
+      expect(navigationTapEl).to.exist;
+    });
+
+    it('should not show navigation screen on viewer response', async () => {
+      env.sandbox.stub(viewer, 'sendMessageAwaitResponse').resolves({
+        screens: [{screen: 'on', show: false}],
+      });
+      env.sandbox
+        .stub(storyEducation.getAmpDoc(), 'whenFirstVisible')
+        .resolves();
+
+      storyEducation.buildCallback();
+      await Promise.resolve(); // whenFirstVisible icrotask tick.
```

Comment:
nit: s/icrotask/microtask

Reply:
Trying to fix this with the "suggestion" feature 👀